djangoProject/users/serializers.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import serializers
+from .models import User
+from django.core.exceptions import ValidationError
+import re
+
+class UserRegistrationSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(write_only=True, min_length=8, required=True)
+    confirm_password = serializers.CharField(write_only=True, required=True)
+
+    class Meta:
+        model = User
+        fields = ['user_type', 'email', 'password', 'confirm_password']
+    
```

Comment:
I didn't find a custom user model in the project but you are already trying to use from .models import User. Please check if your code can store the user in the database.

Please check if the fields in your user model match the fields specified in the database schema.

Reply:
I have added user model and checked if the code stores newly created users successfully in second commit